src/delta_utils/check_submission.py

```diff
@@ -0,0 +1,179 @@
+        f"if __name__ == '__main__': at the bottom of the file"
+    )
+
+    # Check the choose_move() function exists
+    try:
+        choose_move = getattr(mod, "choose_move")
+    except AttributeError as e:
+        raise Exception(
+            f"No function 'choose_move()' found in file {file_name}.py"
+        ) from e
+
+    # Check there is a TEAM_NAME attribute
+    try:
+        team_name = getattr(mod, "TEAM_NAME")
+    except AttributeError as e:
+        raise Exception(f"No TEAM_NAME found in file {file_name}.py") from e
+
+    # Check TEAM_NAME isn't empty
+    if len(team_name) == 0:
+        raise ValueError(f"TEAM_NAME is empty in file {file_name}.py")
+
+    # Check TEAM_NAME isn't still 'Team Name'
+    if team_name == "Team Name":
+        raise ValueError(
+            f"TEAM_NAME='Team Name' which is what it starts as - "
+            f"please change this in file {file_name}.py to your team name!"
+        )
+
+    if (
+        pkl_file is not None
+        and expected_pkl_type is not None
+        and pkl_checker_function is not None
+    ):  # lol mypy
+        try:
+            assert isinstance(
+                pkl_file, expected_pkl_type
+            ), f"The .pkl file you saved is the wrong type! It should be a {expected_pkl_type}"
+            pkl_checker_function(pkl_file)
+            action = choose_move(example_state, pkl_file)
+        except FileNotFoundError as e:
+            raise FileNotFoundError(
+                f"Value dictionary file called 'dict_{team_name}.pkl' cannot be found! "
+                f"Check the file exists & that the name matches."
+            ) from e
+    else:
+        action = choose_move(example_state)
+
+    assert isinstance(action, expected_choose_move_return_type), (
+        f"Action output by `choose_move()` must be type {expected_choose_move_return_type}, "
+        f"but instead {action} of type {type(action)} was output."
+    )
+    print(
+        "Congratulations! Your Repl is ready to submit :)\n\n"
+        f"It'll be using value function file called 'dict_{team_name}.pkl'"
+    )
+
+
+def sha256_file(filename: Path) -> str:
+    h = hashlib.sha256()
+    b = bytearray(128 * 1024)
+    mv = memoryview(b)
+    with open(filename, "rb", buffering=0) as f:
+        while n := f.readinto(mv):  # type: ignore
+            h.update(mv[:n])
+    return h.hexdigest()
+
+
+def pkl_checker_value_dict(pkl_file: Dict) -> None:
+    """Checks a dictionary acting as a value lookup table."""
+    if isinstance(pkl_file, defaultdict):
+        assert not callable(
+            pkl_file.default_factory
+        ), "Please don't use functions within default dictionaries in your pickle file!"
+
+    assert len(pkl_file) > 0, "Your dictionary is empty!"
+
+    for v in pkl_file.values():
+        assert isinstance(
+            v, (float, int)
+        ), "Your value function dictionary values should be a float!"
```

Comment:
Think this is a better error message :smile: 

```suggestion
    for key, v in pkl_file.items():
        assert isinstance(
            v, (float, int)
        ), f"Your value function dictionary values should all be numbers, but for key {key}, the value {v} is of type {type(v)}!"
```